questions.md

```diff
@@ -4,26 +4,125 @@
+
+```
+/usr/bin/time -v commandgoeshere
+```
+
+Which outputs:
+
+```
+* User time (seconds): 72.76
+      * System time (seconds): 3.28
+      * Percent of CPU this job got: 93%
+      * Elapsed (wall clock) time (h:mm:ss or m:ss): 1:21.48
+        Average shared text size (kbytes): 0
+        Average unshared data size (kbytes): 0
+        Average stack size (kbytes): 0
+        Average total size (kbytes): 0
+      * Maximum resident set size (kbytes): 3745120
+        Average resident set size (kbytes): 0
+      * Major (requiring I/O) page faults: 0
+      * Minor (reclaiming a frame) page faults: 233595
+        Voluntary context switches: 17852
+        Involuntary context switches: 24019
+      * Swaps: 0
+      * File system inputs:   496560
+      * File system outputs: 2878576
+      * Socket messages sent: 0
+      * Socket messages received: 0
+        Signals delivered: 0
+        Page size (bytes): 4096
+      * Exit status: 0
+```
+  
+This lists out information on the run, including the user & system time; the maximum resident set size which shows how much RAM is used; and output size which relates to how much data it is writing.
+
 3. In order to be a "good citizen", you need to have some idea of much RAM your job requires.  In particular, you need
    to know the "peak" (i.e., maximum) RAM required at any point during execution.  Show an example of the shell command
-   that you would use on a Linux machine to measure run time and peak ram usage of an arbitrary command, where the time/peak RAM values are written to a file.
+   that you would use on a Linux machine to measure run time and peak ram usage of an arbitrary command, where the time/peak        RAM values are written to a file.
+
+Could run your application/command on a small set of input data/subset of input data and scale up from there, prefixed by:
+
+```
+/usr/bin/time -v commandhere 2>&1 outfile.txt
+```
+
+Based on the values given and the initial size of your subsetted input file vs the total size of the original file you could estimate the time and RAM required for the actual job.
+
+   
 4. What are the units of your answer for number 3?
+
+User and system time is reported in seconds
+
+Elapsed wall clock time is hour:min:sec
+
+Maximum resident set size values are reported in kbytes
+
 5. What are the bash commands for the following operations:
 
     * Checking that a file exists
+    
+    ```
+if [ -e txt.md ]
+then
+    echo "Exists"
+else
+    echo "Does not exist"
+fi
+```
+    
     * Checking that a file exists and is not empty
+    
+    ```
+if [ -e test.md ]
+then
+  if [ -s test.md ]
+  then
+   echo "Exists and is full"
+   else
+    echo "Exists but is empty"
+fi
+else
+  echo "Does not exist"
+fi
+```
 
 6. How would you use the commands from your answer to 5 to write a work flow for HPC that only runs a job if the
    expected output file is **not** present.
 
+```
+if [ -e test.txt ]
```

Comment:
Hi here you could just do if [ ! -e test.txt ] and avoid the else statement as this check if something is not. typically in CS ! means not. Your answer is still correct though. you could also just do [ -e test.txt ] || CMD and make it a one liner